src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -395,13 +395,15 @@ static inline int MPIDI_OFI_handle_lmt_ack(MPIDI_OFI_am_header_t * msg_hdr)
     MPIR_Request *sreq;
     MPIDI_OFI_ack_msg_payload_t *ack_msg;
     int handler_id;
+    int ofi_version = MPIDI_OFI_get_ofi_version();
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_HANDLE_LMT_ACK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_HANDLE_LMT_ACK);
 
     ack_msg = (MPIDI_OFI_ack_msg_payload_t *) (msg_hdr + 1);
     sreq = ack_msg->sreq_ptr;
 
-    if (MPIDI_OFI_ENABLE_MR_SCALABLE) {
+    if ((ofi_version < FI_VERSION(1, 5) && MPIDI_OFI_ENABLE_MR_SCALABLE) ||
+        (ofi_version >= FI_VERSION(1, 5) && !MPIDI_OFI_ENABLE_MR_PROV_KEY)) {
```

Comment:
Since we are adding variable condition, the original benefit of potential configure  constant of `MPIDI_OFI_ENABLE_MR_SCALABLE` disappeared. Then we might as well consolidate this condition into a new variable that is set during init time, for cleaner code.

Reply:
Of course `MR_SCALABLE` is not useful anymore. It's deprecated. 😄 

This probably could be consolidated into a couple of variables though. I'll work on that.

Reply:
Running this through the tests now, but the last patch should take care of this concern @hzhou.